anoma/juvix
# Implement an html documentation generator similar to haddock (#1413)

The 1st iteration of the design of juvix documentation is almost a copy of haddock with the linuwial theme. I've changed some colors following the palette that juvix.org uses in the landing page
Currently the generated documentation consists of:
1. index with the tree of all modules reachable from the main module (which modules show up should be refined in the future)
2. A page for each module with its signatures documented and linked identifiers
3. Pretty printed source code of each module with linked identifiers

To test it
```
cd examples/milestone/MiniTicTacToe
juvix internal doc --open MiniTicTacToe.juvix
```
Note that the `open` flag will not work on Mac since it uses `xdg-open` internally. This will need to be fixed in a later PR

Next I think it would be a good idea to replace the existing `html` command with the new `doc` command
